src/app/institutional/pages/institutional/institutional.component.ts

```diff
@@ -113,11 +118,69 @@ export class InstitutionalComponent implements OnInit {
     this.institution = institutionSelected
     this.entityID = this.institution.entityID
 
-    if (this.institution.Logos) {
+    if (this.institution.Logos && !this.isInstitutionLogoDisplayed(this.institution.Logos[0].value)) {
       this.logoInstitution = this.institution.Logos[0].value
     }
   }
 
+  getNameInstitution(institution: Institutional): DisplayName {
+    return institution.DisplayNames?.find(
+      (name) => name.lang === 'en')
+  }
+
+  addUserSelectedIdPs() {
+    let dataCookie = this._cookie.get('_saml_idp')
+    let institutions = []
+    if (dataCookie) {
+      dataCookie = dataCookie.replace(/^\s+|\s+$/g, '')
+      dataCookie = dataCookie.replace('+', '%20')
+      institutions = dataCookie.split('%20')
+    }
+
+    const dateCookie = new Date(
+      new Date().getTime() + this.cookieExpirationTime * 24 * 60 * 60 * 1000
+    )
+    if (!institutions.includes(btoa(this.entityID))) {
+      if (institutions.length === 3) {
+        institutions = []
```

Comment:
Would this remove all the institutions once the maximum amount is reached? 

Wouldn't be better to delete just one? maybe with a FIFO approach

Reply:
Yes, It would remove all. Ok, I will update it